torch/csrc/autograd/variable.h

```diff
@@ -608,7 +607,7 @@ inline Variable make_variable(
       !data.is_variable(),
       "Must not create a new variable from a variable, use its .data()");
   if (data.defined()) {
-    auto data_impl_copy = data.getIntrusivePtr()->shallow_copy_and_detach();
+    auto data_impl_copy = data.getIntrusivePtr()->shallow_copy_and_detach(/*version_counter=*/0);
     data_impl_copy->set_allow_tensor_metadata_change(allow_tensor_metadata_change);
```

Comment:
are there cases where we need to not `set_allow_tensor_metadata_change` and then set it later?  Or is it always "set" correctly at the `shallow_copy_and_detach` step?

Reply:
It's always "set" correctly at the `shallow_copy_and_detach` step except for this one case: https://github.com/pytorch/pytorch/blob/da3e74b21c94d24175a5f1cd6d7f2f049f16ca9e/torch/csrc/autograd/variable.h#L587-L601

I think we should add `allow_tensor_metadata_change` as another parameter into `shallow_copy_and_detach()`, to zip up this API further.

Reply:
Update: I added the `allow_tensor_metadata_change` parameter to `shallow_copy_and_detach()`.